doc/policy/packages.md

```diff
@@ -117,3 +118,30 @@ backward-compatible for users and applications that rely on p2p transaction rela
 package validation should not prevent the acceptance of a transaction that would otherwise be
 policy-valid on its own. By always accepting a transaction that passes individual validation before
 trying package validation, we prevent any unintentional restriction of policy.
+
+### Package Replace by Fee
+
+Child-with-unconfirmed-parents packages may replace mempool transactions under the following conditions:
+
+1. The child transaction does not conflict with any transactions in the mempool.
+
+2. The directly conflicting transactions all signal replaceability explicitly.
+
+3. All transactions in the package, including those that do not directly conflict with any mempool
+   transactions, only include an unconfirmed input if that input was included in one of the directly
+   conflicting transactions or is from another transaction in the package.
```

Comment:
My head hurts :laughing: 

I'm not 100% sure I'm understanding this rule correctly, can you let me know if the following example makes sense?

Let's imagine we have `{commitTxAlice, anchorTxAlice}` in the mempool that we want to replace.
What we really want to do is to replace `commitTxAlice` by `commitTxBob` (the child is just a tool to achieve that).
We ask `bitcoind` to `fundrawtransaction` to ensure our `anchorTxBob` adds enough funds.
`bitcoind` adds an unconfirmed (but safe) wallet input:

```
+-------------+
| commitTxBob |---------+
+-------------+         |      +-------------+
                        |      |             |
+------------------+    +----->| anchorTxBob |
| previousWalletTx |---------->|             |
+------------------+           +-------------+
```

If we include `previousWalletTx` in the package, this package-rbf would work, right?
But if we don't include `previousWalletTx` in the package, this would be rejected?

But what if `previousWalletTx` itself has an unconfirmed parent?
We cannot include this grand-parent in the package (since packages are only parents-with-single-child).

This isn't directly bringing an unconfirmed input, but it is indirectly bringing one. Is that ok? If that's ok, why wouldn't it be ok to just allow adding new unconfirmed inputs in the first place?

Reply:
Unfortunately, no, `previousWalletTx` cannot be used as one of the parents; all of `anchorTxBob`'s inputs need to be confirmed.

This is an extension of BIP 125 Rule 2. I currently don't know a safe way to remove that rule. #23121 was an attempt, but also unsafe. Very sorry :(

Reply:
Gotcha, thanks for the confirmation, we'll deal with that limitation then, it's already a good start. And hopefully we'll eventually figure out better RBF rules for both single txs and packages!

Reply:
(Sorry for my earlier comment, I was confused)

> If we include previousWalletTx in the package, this package-rbf would work, right?

Yes. It's fine because it's one of the direct parents of `anchorTxBob`.

> But if we don't include previousWalletTx in the package, this would be rejected?

If you mean `anchorTxBob` now only has 1 unconfirmed input, `commitTxBob`, actually this would be accepted.

If you mean that `previousWalletTx` was submitted ahead of time, that should still be okay because the code will remember to exempt `previousWalletTx` from the "no new unconfirmed" rule.

> But what if previousWalletTx itself has an unconfirmed parent?

Based on this rule, if `previousWalletTx` has another unconfirmed parent, this would be rejected.

Reply:
Thanks for the clarification! I'll create extensive tests of this behavior and will report back.

Reply:
It's looking good with the current state of the PR, I am able to include new unconfirmed inputs that themselves have an unconfirmed parent as shown by [this test](https://gist.github.com/t-bast/7c553e61ff2bee3720ff4f7db04cc1b3#batching-commitment-fee-bump) :+1:

Reply:
You will have to find the best way to go about making it understanding for you it's like c++ not everyone nos how to use it so they find the eazyist way for them selfs to move on iam Satoshi an it's been 13 years I have not work on aney thing so it's all new to me agin but we will make do an move on let other no I'm back not to change anything or not to be in the spot light but I'm am alive an people need to stop trying to break in to my address with my Bitcoin I will never make Bitcoin go under an inflashion can disrupt the hole world an I won't peace in the world.